### PR TITLE
Ticket_#3519 QueryDict method doc string update.

### PR DESCRIPTION
**Pull request by jburns6789**
Updated doc string to include the return of of [] when a key references an empty list.

ticket-35915 Trac ticket number

ticket-35915

#### Branch description
Provide a concise overview of the issue or rationale behind the proposed changes.

#### Checklist
- [ ] This PR targets the `main` branch. 
- [ ] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [ ] I have checked the "Has patch" ticket flag in the Trac system.
- [ ] I have added or updated relevant tests.
- [ ] I have added or updated relevant docs, including release notes if applicable.
- [ ] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: docs/ref/request-response.txt [hunk 1]
```diff
@@ -554,12 +554,11 @@ a subclass of dictionary. Exceptions are outlined here:
...
-    Returns the value for the given key. If the key has more than one value,
-    it returns the last value. Raises
-    ``django.utils.datastructures.MultiValueDictKeyError`` if the key does not
-    exist. (This is a subclass of Python's standard :exc:`KeyError`, so you can
-    stick to catching ``KeyError``.)
-
+    Returns the last data value for the given key. If the key has more than one value,
+    it returns the last value. If the value is an empty list, ``[]`` is returned. 
+    Raises ``django.utils.datastructures.MultiValueDictKeyError`` if the key does not exist. 
+    (This is a subclass of Python's standard :exc:`KeyError`, so you can stick to catching ``KeyError``.)
```
Comment (adamzap) [inline docs/ref/request-response.txt:560]: Hello! Docs should be wrapped at 80 characters: https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#guidelines-for-restructuredtext-files

**Comment by adamzap** [comment]
The line wrapping looks good to me! I made one more comment.

Once you feel that this PR is ready, you can mark the "Has patch" field in Trac. Then someone with more authority will eventually review it for content.

File: docs/ref/request-response.txt [hunk 2]
```diff
@@ -554,12 +554,12 @@ a subclass of dictionary. Exceptions are outlined here:
...
-    ``django.utils.datastructures.MultiValueDictKeyError`` if the key does not
-    exist. (This is a subclass of Python's standard :exc:`KeyError`, so you can
-    stick to catching ``KeyError``.)
-
+    Returns the last data value for the given key. If the key has more than one
+    value, it returns the last value. If the value is an empty list, ``[]`` is
+    returned. Raises ``django.utils.datastructures.MultiValueDictKeyError`` if
+    the key does not exist. (This is a subclass of Python's standard
+    :exc:`KeyError`, so you can stick to catching ``KeyError``.)
+
```
[the remark this answers is not in the document]
Comment (jburns6789) [reply docs/ref/request-response.txt:562]: appears so, i will remove

**Comment by jburns6789** [comment]
Thank You

**Comment by adamzap** [comment]
@jburns6789 It looks like Trac is not associating this PR with the ticket. Could you try changing the beginning of the Markdown in the initial post of this PR to this?

```markdown
#### Trac ticket number

ticket-35915
```

---

You can also fix the title of this PR by fixing the ticket number and making the title (and commit message) "Ticket #35915 -- QueryDict method doc string update."

https://docs.djangoproject.com/en/dev/internals/contributing/committing-code/#committing-guidelines

**Comment by shangxiao** [comment]
Hi jburns, thanks for the patch 🏆 

This one's hard to concisely describe due to its quirky nature.

These 2 sentences here are repetitive `Returns the last data value for the given key. If the key has more than one value, it returns the last value.`

Also I found this part here `If the value is an empty list, ``[]`` is returned` confusing even though that's what the docstring says.

I'll leave this suggestion here for Sarah or Nessita to consider:

> Retrieves the collection of values for the given key and returns the last item in the collection; or an empty list (``[]``) if the collection is empty.  Raises ... etc 

I'm not sure collection is the right word to use but just trying to avoid confusing folks by mentioning lists twice. 🤷

**Comment by jburns6789** [comment]
Makes sense, I think the more concise the better based on the nature of the behavior, the doc string now mentions four cases instead of three.

**Comment by jburns6789** [comment]
I am also working in changing the markdown and commit message

**Comment by jburns6789** [comment]
updated the title and added the ticket trac number https://github.com/django/django/pull/18841

**Pull request closed without merging**